ios/NativeModules/TextileNode.m

```diff
@@ -47,6 +48,77 @@ - (dispatch_queue_t)methodQueue {
   return dispatch_queue_create("io.textile.TextileNodeQueue", DISPATCH_QUEUE_SERIAL);
 }
 
+// Export method for local photo selection
+
+RCT_EXPORT_METHOD(requestLocalPhotos:(int)minEpoch resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
+  NSTimeInterval seconds = minEpoch;
+  NSDate *epochNSDate = [[NSDate alloc] initWithTimeIntervalSince1970:seconds];
+
+  PHAuthorizationStatus status = [PHPhotoLibrary authorizationStatus];
+  if (status == PHAuthorizationStatusAuthorized)
+  {
+    // Setup date-string conversion
+    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+    // Ensure date string is always in UTC
+    [dateFormatter setTimeZone:[NSTimeZone timeZoneWithAbbreviation:@"UTC"]];
+    [dateFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"];
+
+    //initialize empty local state
+    @autoreleasepool
+    {
+      //fetch all the albums
+      PHFetchOptions *onlyImagesOptions = [PHFetchOptions new];
+
+      // Limit to only photo media types
+      // NSPredicate *mediaPhotos = [NSPredicate predicateWithFormat:@"mediaType = %i", PHAssetMediaTypeImage];
+      // Limit to only photos modified later than or equal to our specified date
+      NSPredicate *minDate = [NSPredicate predicateWithFormat:@"modificationDate > %@", epochNSDate];
+      // Limit to camera roll?
+      // NSPredicate *cameraRoll = [NSPredicate predicateWithFormat:@"modificationDate > %@", epochNSDate];
+      onlyImagesOptions.predicate = minDate;
+      // Combine the multiple predicates (didn't seem to work combining them like this)
+      // [NSCompoundPredicate orPredicateWithSubpredicates:@[mediaPhotos, minDate]];
+      // Not sure we really need to do the sort... but here it is
+      onlyImagesOptions.sortDescriptors = @[[NSSortDescriptor sortDescriptorWithKey:@"modificationDate" ascending:YES]];
+
+
+      PHFetchResult *allPhotosResult = [PHAsset fetchAssetsWithMediaType:PHAssetMediaTypeImage options:onlyImagesOptions];
+      [allPhotosResult enumerateObjectsUsingBlock:^(PHAsset *asset, NSUInteger idx, BOOL *stop) {
+
+        // Get our path right away while we are in Obj-c
+        [asset requestContentEditingInputWithOptions:nil completionHandler:^(PHContentEditingInput * _Nullable contentEditingInput, NSDictionary * _Nonnull info) {
+
+          // get rid of file://
+          NSString *path = [contentEditingInput.fullSizeImageURL.absoluteString substringFromIndex:7];
```

Comment:
Is this now a path that we "own" and are responsible for cleaning up? I think so, just want to get your take on it @andrewxhill

Reply:
I was under the impression that it was not something we needed to clean...